scipy/optimize/zeros.py

```diff
@@ -157,12 +170,15 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
         raise ValueError("tol too small (%g <= 0)" % tol)
     if maxiter < 1:
         raise ValueError("maxiter must be greater than 0")
+    if not np.isscalar(x0):
+        return _array_newton(func, x0, fprime, args, tol, maxiter, fprime2,
+                             **kwargs)
     # Multiply by 1.0 to convert to floating point.  We don't use float(x0)
     # so it still works if x0 is complex.
     p0 = 1.0 * x0
```

Comment:
Shouldn't this multiplying by 1.0 happen before the call to `_array_newton`? Or is there a reason it's needed for scalars but not for arrays?

Reply:
✔️